drivers/lora/sx12xx_common.c

```diff
@@ -22,14 +22,18 @@
 
 LOG_MODULE_REGISTER(sx12xx_common, CONFIG_LORA_LOG_LEVEL);
 
+struct sx12xx_rx_params {
+	uint8_t *buf;
+	uint8_t *size;
```

Comment:
`size` need not be a pointer

Reply:
It does need to be a pointer, as this is how the caller knows how many bytes were actually received.

Reply:
You could just return `dev_data.rx_params.size` directly from `sx12xx_lora_recv` instead of `size`. Moreover, we should not try to modify the API arguments without the knowledge of the application.

Reply:
We cannot just return `dev_data.rx_params.size` directly as the data struct is no longer owned by the caller at that point.
The API arguments are not being modified here. The input variable is still a value, not a pointer.
I'm just using the existing size variable instead of creating an additional temporary.

Reply:
What I'm trying to say is, you are taking the address of the input argument `size` and storing the actual size in it in case if it differs. So once the `recv` call returns, both `size` argument and the return value will be the same. My argument is, if the API declares a function argument as non-pointer, the API definition should not modify its value.

Reply:
This can be fixed easily if `dev_data.rx_params.size` is made non-pointer. So it can copy the input `size` value and during the end if `recv`, its value can be returned.

Reply:
> both size argument and the return value will be the same

That is not the case. Arguments passed to a function by value will never have a different value upon returning from that function. The value is always a copy, as a random reference https://denniskubes.com/2012/08/20/is-c-pass-by-value-or-reference/.

> So it can copy the input size value and during the end if recv, its value can be returned.

My original comment on this was wrong, but not in a way that makes this possible.
As soon as the `sx12xx_ev_rx_done` callback finishes running, the original caller to `lora_recv` no longer has ownership of the `rx_params` struct. Therefore even immediately after returning from `k_poll`, we can make no guarantees about the value of any parameter that it contains. For a value to make its way back to the `lora_recv` caller, it is mandatory for that value to be a pointer to a stack variable which is then populated in the callback itself.

Reply:
Its no different from
```
int square(int x) {
    x *= x;
    return x;
}

int main(void) {
    int a = 5;
    int sq = square(a);
    printk("%d, %d\n",  a, sq);
}
```
It will print out "5, 25", not "25 25"

Reply:
doh.. Sorry, my comment about the `size` argument is wrong (Brain fade). But still, please use a separate local variable to copy the `size` value as I still feel that it is not good to take reference of the function argument.

But if you don't wanna do that I'm fine too. Don't want to stretch this discussion too much!